src/esp/scene/SceneNode.cpp

```diff
@@ -3,35 +3,191 @@
 // LICENSE file in the root directory of this source tree.
 
 #include "SceneNode.h"
+#include "SceneGraph.h"
+#include "esp/core/Check.h"
 #include "esp/geo/geo.h"
+#include "esp/sensor/Sensor.h"
 
 namespace Mn = Magnum;
 
 namespace esp {
 namespace scene {
 
-SceneNode::SceneNode(SceneNode& parent)
-    : Mn::SceneGraph::AbstractFeature3D{*this} {
-  setParent(&parent);
-  setId(parent.getId());
+SceneNode::SceneNode() : Mn::SceneGraph::AbstractFeature3D{*this} {
```

Comment:
Wait. Just realize this is a new constructor. Is it a must have?
The consequence is that with this default constructor, the user can now create a scene node without connecting it to any  scene graph. (dangling scene node).

Reply:
OK. I see the code below. It is not a public constructor. But again, is it a must have?